### PR TITLE
Add a mechanism to add notes on specific DNS records; use this for Hetzner CNAME records

### LINKED ISSUE #148: Unable to create CNAME entry with hetzner_dns_record_set
<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Also test if the latest release and devel branch are affected too -->
<!--- Complete *all* sections as described, this form is processed automatically -->

##### SUMMARY
<!--- Explain the problem briefly below -->
Unable to create CNAME entry with hetzner_dns_record_set

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below, use your best guess if unsure -->
hetzner_dns_record_set

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible [core 2.15.0]
  config file = /home/jxxx/Development/own/infrastructure/ansible.cfg
  configured module search path = ['/home/xxx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.11/site-packages/ansible
  ansible collection location = /home/xxx/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.11.3 (main, Jun  5 2023, 09:32:32) [GCC 13.1.1 20230429] (/usr/bin/python)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION
<!--- Paste verbatim output from "ansible-galaxy collection list <namespace>.<collection>"  between the quotes
for example: ansible-galaxy collection list community.general
-->
```paste below

# /usr/lib/python3.11/site-packages/ansible_collections
Collection    Version
------------- -------
community.dns 2.5.4 
```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below
CONFIG_FILE() = /home/xxx/Development/own/infrastructure/ansible.cfg
EDITOR(env: EDITOR) = /usr/bin/vim
RETRY_FILES_ENABLED(/home/xxx/Development/own/infrastructure/ansible.cfg) = True
RETRY_FILES_SAVE_PATH(/home/xxx/Development/own/infrastructure/ansible.cfg) = /home/xxx/.ansible/retry-files
```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. target OS versions, network device firmware, etc. -->
Linux (ArchLinux)

##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->

Create a playbook with two tasks: The first task creates a new A record and the second tasks creates a CNAME record.

<!--- Paste example playbooks or commands between quotes below -->
```yaml
---                                                                                                                                                            
- hosts: localhost
  gather_facts: no

  tasks:
    - community.dns.hetzner_dns_record_set:
        hetzner_token: access_token
        state: present
        zone_name: example.org
        type: A
        record: a.example.org
        value: 1.1.1.1
        ttl: 86400

    - community.dns.hetzner_dns_record_set:
        hetzner_token:  access_token
        state: present
        zone_name: example.org
        type: CNAME
        record: a.example.org
        value: b.example.org
        ttl: 86400
```

<!--- HINT: You can paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!--- Describe what you expected to happen when running the steps above -->
A new CNAME entry is created with "b.example.org" and the value "a".

##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->

The Hetzner DNS API returns with error 422 Unprocessable Entity.

<!--- Paste verbatim command output between quotes -->
```paste below
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: xxx                                                    
<127.0.0.1> EXEC /bin/sh -c 'echo ~xxx && sleep 0'                                                                                                                                                                                                           
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/xxx/.ansible/tmp `"&& mkdir "` echo /home/xxx/.ansible/tmp/ansible-tmp-1687327611.7306418-67081-28898446076627 `" && echo ansible-tmp-1687327611.7306418-67081-28898446076627="` echo /home/xxx/.ansible/tmp/ansible-tmp-1687327611.7306418-67081-28898
446076627 `" ) && sleep 0'                                                                                                                                                                                                                                     
Using module file /usr/lib/python3.11/site-packages/ansible_collections/community/dns/plugins/modules/hetzner_dns_record_set.py 
<127.0.0.1> PUT /home/xxx/.ansible/tmp/ansible-local-67060udn51gzq/tmpj4yotnxx TO /home/jenne/.ansible/tmp/ansible-tmp-1687327611.7306418-67081-28898446076627/AnsiballZ_hetzner_dns_record_set.py                                                                                                                           
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/xxx/.ansible/tmp/ansible-tmp-1687327611.7306418-67081-28898446076627/ /home/xxx/.ansible/tmp/ansible-tmp-1687327611.7306418-67081-28898446076627/AnsiballZ_hetzner_dns_record_set.py && sleep 0'              
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/xxx/.ansible/tmp/ansible-tmp-1687327611.7306418-67081-28898446076627/AnsiballZ_hetzner_dns_record_set.py && sleep 0'                                                                                                                                                      
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/xxx/.ansible/tmp/ansible-tmp-1687327611.7306418-67081-28898446076627/ > /dev/null 2>&1 && sleep 0'                                                                                                               
The full traceback is:                                                                                                                                         
Traceback (most recent call last):                                                                                                                                                                                   
  File "/tmp/ansible_community.dns.hetzner_dns_record_set_payload_ffclvt5f/ansible_community.dns.hetzner_dns_record_set_payload.zip/ansible_collections/community/dns/plugins/module_utils/module/record_set.py", line 247, in run_module                      
    raise errors[0]                                                                                                                                            
  File "/tmp/ansible_community.dns.hetzner_dns_record_set_payload_ffclvt5f/ansible_community.dns.hetzner_dns_record_set_payload.zip/ansible_collections/community/dns/plugins/module_utils/zone_record_helpers.py", line 124, in bulk_apply_changes                                                                            
    record = api.add_record(zone_id, record)                                                                                                                                                                         
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                         
  File "/tmp/ansible_community.dns.hetzner_dns_record_set_payload_ffclvt5f/ansible_community.dns.hetzner_dns_record_set_payload.zip/ansible_collections/community/dns/plugins/module_utils/hetzner/api.py", line 204, in add_record     
    raise DNSAPIError(                 
ansible_collections.community.dns.plugins.module_utils.zone_record_api.DNSAPIError: The new CNAME record with value "b.example.org" and TTL 86400 has not been accepted by the server with error message "422 Unprocessable Entity: taken  a" (error code 422)                                                                
fatal: [localhost]: FAILED! => {       
    "changed": false,                                                                                                                                                                                                                   
    "error": "The new CNAME record with value \"b.example.org\" and TTL 86400 has not been accepted by the server with error message \"422 Unprocessable Entity: taken  a\" (error code 422)",                                         
    "invocation": {                                                                                                                                                                                                  
        "module_args": {                                                                                  
            "bulk_operation_threshold": 2,                                                                                                                                                                           
            "hetzner_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",                                       
            "on_existing": "replace",                                                                                                                                                                                
            "prefix": null,                                                    
            "record": "a.example.org",                                                                                                                                                                                                 
            "state": "present",                                                                                     
            "ttl": 86400,                                                                                                                                                                                            
            "txt_character_encoding": null,                                                                                                                                                                                                                    
            "txt_transformation": "unquoted",                                                                                                                                                                                           
            "type": "CNAME",                                                                                                                                                                                                                                   
            "value": [                               
                "b.example.org"                                                                                                                                                                                                                               
            ],                                                                                                                                                                                                                          
            "zone_id": null,                                                                                        
            "zone_name": "example.org"                                                                                                                                                                              
        }                                                                                                                                                                                                                                                      
    },                                                                                                    
    "msg": "Error: The new CNAME record with value \"b.example.org\" and TTL 86400 has not been accepted by the server with error message \"422 Unprocessable Entity: taken  a\" (error code 422)"  
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #148.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
DNS record modules

### CODE REVIEW

File: plugins/doc_fragments/hetzner.py [hunk 1]
```diff
@@ -33,6 +33,15 @@ class ModuleDocFragment(object):
           - name: HETZNER_DNS_TOKEN
 '''
 
+    # NOTE: This document fragment adds additional information on records.
+    RECORD_NOTES = r'''
+options: {}
+notes:
+  - For C(CNAME) records, use absolute DNS names for values. Absolute DNS names are DNS names ending with
+    a period C(.), for example C(foo.example.com.). If you use a relative DNS name, that is one without a
+    trailing period, it will be treated relative to the zone of the C(CNAME) record.
```
Comment (oraNod) [inline plugins/doc_fragments/hetzner.py:42]:
```suggestion
  - For C(CNAME) records, use absolute DNS names for values. Absolute DNS names end with
    a trailing period C(.), for example C(foo.example.com.). If you use a relative DNS name, with no
    trailing period, the record will be relative to the zone of the C(CNAME) record.
```

**Review by oraNod — COMMENTED** [review]
Minor nits for clarity.

File: plugins/doc_fragments/hetzner.py [hunk 2]
```diff
@@ -33,6 +33,15 @@ class ModuleDocFragment(object):
           - name: HETZNER_DNS_TOKEN
 '''
 
+    # NOTE: This document fragment adds additional information on records.
+    RECORD_NOTES = r'''
+options: {}
+notes:
+  - For C(CNAME) records, use absolute DNS names for values. Absolute DNS names end with
+    a trailing period C(.), for example C(foo.example.com.). If you use a relative DNS name, with no
+    trailing period, the record will be relative to the zone of the C(CNAME) record.
```
Comment (felixfontein) [inline plugins/doc_fragments/hetzner.py:42]:
```suggestion
    trailing period, the value will be relative to the zone of the C(CNAME) record.
```

**Review by oraNod — APPROVED** [review]

**Comment by felixfontein** [comment]
@oraNod thanks a lot!

**Pull request merged**